### PR TITLE
Find next free event

### PR DESCRIPTION
**Pull request by Pacane**

### CODE REVIEW

File: src/main/java/org/dsa/iot/calendar/event/EventUtils.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                    if (MINUTES.between(o1.getStart(), now) > MINUTES.between(o2.getStart(), now)) {
+                        return 1;
+                    } else if (MINUTES.between(o1.getStart(), now) < MINUTES.between(o2.getStart(), now)) {
+                        return -1;
+                    } else {
+                        return 0;
+                    }
+                }).collect(toList());
+
+        for (int i = 0; i < events.size() - 1; ++i) {
```
Comment (kubeliv) [inline src/main/java/org/dsa/iot/calendar/event/EventUtils.java]: Should this be using eventsNotOver?

File: build.gradle [hunk 2]
```diff
@@ -36,7 +36,8 @@ dependencies {
 
     // Exchange Web Services Support
     compile 'com.microsoft.ews-java-api:ews-java-api:2.0'
-    compile 'com.arcbees:checkstyle:1.2'
+    compile group: 'junit', name: 'junit', version: '4.12'
```
Comment (azenla) [inline build.gradle]: Probably should use the same 'A:B:C' syntax here

File: src/main/java/org/dsa/iot/calendar/caldav/CalDAVCalendar.java [hunk 3]
```diff
@@ -47,7 +45,9 @@ public CalDAVCalendar(String host, int port, String path, Node eventsNode) {
 
     @Override
     public void createEvent(DSAEvent event) {
-        VEvent vEvent = new VEvent(new Date(event.getStart().getTime()), new Date(event.getEnd().getTime()), event.getTitle());
+        java.util.Date start = Date.from(event.getStart());
```
Comment (azenla) [inline src/main/java/org/dsa/iot/calendar/caldav/CalDAVCalendar.java:48]: Was this expanded to java.util.Date for a reason? Conflicting imports?

File: src/test/java/org/dsa/iot/calendar/event/TimeSlotTableTest.java [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+        TimeRange timeRange = new TimeRange(getInstantFromString("2016-10-24T10:00"), getInstantFromString("2016-10-24T11:00"));
+
+        timeSlotTable.mergeSlot(timeRange);
+        timeSlotTable.mergeSlot(timeRange);
+
+        List<TimeRange> result = timeSlotTable.getTable();
+        assertThat(getHour(result.get(0).start)).isEqualTo(10);
+        assertThat(getHour(result.get(0).end)).isEqualTo(11);
+        assertThat(result).hasSize(1);
+
```
Comment (azenla) [inline src/test/java/org/dsa/iot/calendar/event/TimeSlotTableTest.java]: Empty line

File: src/main/java/org/dsa/iot/calendar/Actions.java [hunk 5]
```diff
@@ -388,44 +383,39 @@ public void handle(ActionResult actionResult) {
...
+                        if (calendar.supportsMultipleCalendars()) {
+                            String calendarId = editEventNode.getChild("calendarId").getValue().getString();
+                            event.setCalendar(new DSAIdentifier(calendarId,
+                                    editEventNode.getChild("calendar").getValue().getString()));
                         }
+                        calendar.deleteEvent(editEventNode.getName(), false);
+                        calendar.createEvent(event);
+                        actionResult.getTable().addRow(Row.make(new Value(true)));
+                    } catch (Exception e) {
+                        e.printStackTrace();
```
Comment (azenla) [inline src/main/java/org/dsa/iot/calendar/Actions.java]: Probably not good to print the stacktrace to stdio, need to use a logger or return error gracefully to the action.

File: src/main/java/org/dsa/iot/calendar/event/EventUtils.java (hunk 1 above)
Comment (Pacane) [reply src/main/java/org/dsa/iot/calendar/event/EventUtils.java]: Yes most likely.  This method was kind of on hold though.  Needed to get the time slot table done to finish it.

File: src/main/java/org/dsa/iot/calendar/Actions.java [hunk 6]
```diff
@@ -321,50 +326,40 @@ public void handle(ActionResult event) {
...
+                            String calendar1 = actionResult.getParameter("calendar").getString();
+                            int indexOfPipe = calendar1.lastIndexOf('|');
+                            String calTitle = calendar1.substring(0, indexOfPipe);
+                            String calUid = calendar1.substring(indexOfPipe + 1);
+                            event.setCalendar(new DSAIdentifier(calUid, calTitle));
                         }
+                        calendar.createEvent(event);
+                        actionResult.getTable().addRow(Row.make(new Value("Event created.")));
+                    } catch (Exception e) {
+                        e.printStackTrace();
```
Comment (azenla) [inline src/main/java/org/dsa/iot/calendar/Actions.java]: Probably not good to print the stacktrace to stdio, need to use a logger or return error gracefully to the action.

File: src/main/java/org/dsa/iot/calendar/Actions.java [hunk 7]
```diff
@@ -388,44 +383,39 @@ public void handle(ActionResult actionResult) {
 
     private static class EditEvent extends Action {
         EditEvent(final BaseCalendar calendar) {
```
Comment (azenla) [inline src/main/java/org/dsa/iot/calendar/Actions.java:381]: Personally find it best to have a modifier here.

**Review by azenla — CHANGES_REQUESTED** [review]
Overall really good.

File: src/main/java/org/dsa/iot/calendar/caldav/CalDAVCalendar.java (hunk 3 above)
Comment (logangorence) [inline src/main/java/org/dsa/iot/calendar/caldav/CalDAVCalendar.java]: Yes, CalDAV has a class called Date too 👿

Comment (kubeliv) [reply src/main/java/org/dsa/iot/calendar/caldav/CalDAVCalendar.java:48]: Yes, CalDAV has a class called Date too 😞

**Comment by Pacane** [comment]
@kaendfinger @logangorence fixed the comments.

**Review by azenla — APPROVED** [review]

**Pull request merged**